docs/source/index.rst

```diff
@@ -81,9 +81,10 @@ the :ref:`api` documentation or alternatively check the `github repository`_.
    installation
    quickstart
    tutorial
-   cli
+   dataset
```

Comment:
I would put the dataset section after the key data model concepts since you're already using some of these undefined concepts in the `dataset` section